src/widgets/settingspages/SettingWidget.cpp

```diff
@@ -227,6 +227,8 @@ template SettingWidget *SettingWidget::dropdown<TabStyle>(
     const QString &label, EnumStringSetting<TabStyle> &setting);
 template SettingWidget *SettingWidget::dropdown<ShowModerationState>(
     const QString &label, EnumStringSetting<ShowModerationState> &setting);
+template SettingWidget *SettingWidget::dropdown<EmojiStyle>(
+    const QString &label, EnumStringSetting<EmojiStyle> &setting);
```

Comment:
We should avoid this...

Reply:
I switched to this in [`5dd92ad` (#6293)](https://github.com/Chatterino/chatterino2/pull/6293/commits/5dd92ad55a8e0abab885eb48b068217bc683eea1) to prevent the template function from being defined inside the header file. imo this ugliness outweighs having the file in the header